src/Vogen/GenerateEqualsMethodsAndOperators.cs

```diff
@@ -144,11 +144,11 @@ public static string GenerateEqualsOperatorsForPrimitivesIfNeeded(string itemUnd
         if (item.Config.PrimitiveEqualityGeneration.HasFlag(PrimitiveEqualityGeneration.GenerateOperators))
             return $"""
                     
-                            public static global::System.Boolean operator ==({typeName} left, {itemUnderlyingType} right) => Equals(left.Value, right);
-                            public static global::System.Boolean operator !=({typeName} left, {itemUnderlyingType} right) => !Equals(left.Value, right);
+                            public static global::System.Boolean operator ==({typeName} left, {itemUnderlyingType} right) => left.Equals(right);
```

Comment:
If the wrapper is a reference type and is null, this will cause a null reference exception. 
I think the code that you've done here is 100% correct for when the wrapper is a value type (and for when the underlying primitives are value types).

Reply:
Right....I somehow assumed I was just dealing with the struct portion of the generator but I don't know what led me to this thinking. I'll revisit this shortly.

Reply:
Looking at it, if `left.Equals` throws an NRE, `left.Value` would also have thrown. I'll restrict my PR to only include the change in `StructGenerator` first and then go from there in a separate PR.

Reply:
My initial changes in `GenerateEqualsMethodsAndOperators` are wrong anyway in case `GenerateOperators` is defined but not `GenerateMethods`.